lib/adal.js

```diff
@@ -810,6 +843,9 @@ var AuthenticationContext = (function () {
                 this.verbose('Window is redirecting');
                 callback = this.callback;
             }
+            callback = this.callback;
```

Comment:
Why set this callback here? I think we need to understand the if checks: window && window.oauth2Callback. That else if block should execute for login requests.